crates/storage/src/db.rs

```diff
@@ -0,0 +1,25 @@
+use crate::{errors::StoreError, redb_interface::ReamDB};
+
+pub struct Database {
+    /// Database for the ream client
+    ///
+    /// Currently this implementation has a single reference to redb instance
+    /// but eventually we would like to add support for multiple backends and
+    /// can be set behind feature flags
+    ///
+    /// `redb::Database` wrapper around with async security features
+    db: ReamDB,
+
+    pub version: String,
+}
+
+/// This implemenation houses all the client<>db interaction logic
+impl Database {
+    pub fn new() -> Result<Self, StoreError> {
+        let db_instance = ReamDB::new()?;
+        Ok(Self {
+            db: db_instance,
+            version: String::from("0.0.1"),
+        })
+    }
```

Comment:
```suggestion
        Ok(Self {
            db: ReamDB::new()?,
            version: String::from("0.0.1"),
        })
    }
```
I'd prefer to use intermediate variables for both or not.